### PR TITLE
csrgeam and csrgeam2

### PR DESCRIPTION
**Pull request by ntrost57**

### CODE REVIEW

File: clients/include/testing_csrgeam2.hpp [hunk 1]
```diff
@@ -0,0 +1,1493 @@
...
+                                                    dBcol,
+                                                    descr_C,
+                                                    dCptr_2,
+                                                    dnnz_C,
+                                                    dbuffer));
+
+        // Compute csrgemm host solution
+        std::vector<int> hcsr_row_ptr_C_gold(M + 1);
+
+        double cpu_time_used = get_time_us();
```
Comment (jsandham) [inline clients/include/testing_csrgeam2.hpp:1370]: Is this used anywhere?

File: clients/common/hipsparse_template_specialization.cpp [hunk 2]
```diff
@@ -1466,6 +1466,526 @@ namespace hipsparse
...
+                                                       const int*                csrRowPtrA,
+                                                       const int*                csrColIndA,
+                                                       const float*              beta,
+                                                       const hipsparseMatDescr_t descrB,
+                                                       int                       nnzB,
+                                                       const float*              csrValB,
+                                                       const int*                csrRowPtrB,
+                                                       const int*                csrColIndB,
+                                                       const hipsparseMatDescr_t descrC,
+                                                       float*                    csrValC,
```
[the remark this answers is not in the document]
Comment (ntrost57) [reply clients/common/hipsparse_template_specialization.cpp:1654]: Should indeed be const. I do not care much about the naming there as it's just the unit tests.

File: clients/include/testing_csrgeam2.hpp (hunk 1 above)
Comment (ntrost57) [reply clients/include/testing_csrgeam2.hpp:1370]: No

File: clients/include/testing_csrgeam2.hpp [hunk 3]
```diff
@@ -0,0 +1,1493 @@
...
+                                                                  dAptr,
+                                                                  dAcol,
+                                                                  descr_B,
+                                                                  safe_size,
+                                                                  dBptr,
+                                                                  dBcol,
+                                                                  descr_C,
+                                                                  dCptr,
+                                                                  nullptr,
+                                                                  dbuffer),
```
[the remark this answers is not in the document]
Comment (ntrost57) [reply clients/include/testing_csrgeam2.hpp:254]: Yes, should be tested too.

[the remark this answers is not in the document]
Comment (ntrost57) [reply clients/include/testing_csrgeam2.hpp:254]: No, wait, we do not use a buffer in rocsparse, therefore it should not be tested.

**Pull request merged**